arduino-ide-extension/src/browser/boards/boards-service-provider.ts

```diff
@@ -1,37 +1,137 @@
-import { injectable, inject } from '@theia/core/shared/inversify';
-import { Emitter } from '@theia/core/lib/common/event';
-import { ILogger } from '@theia/core/lib/common/logger';
+import { FrontendApplicationContribution } from '@theia/core/lib/browser/frontend-application';
+import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
 import {
   Command,
   CommandContribution,
   CommandRegistry,
   CommandService,
 } from '@theia/core/lib/common/command';
+import type { Disposable } from '@theia/core/lib/common/disposable';
+import { Emitter } from '@theia/core/lib/common/event';
+import { ILogger } from '@theia/core/lib/common/logger';
 import { MessageService } from '@theia/core/lib/common/message-service';
-import { FrontendApplicationContribution } from '@theia/core/lib/browser/frontend-application';
-import { RecursiveRequired } from '../../common/types';
+import { nls } from '@theia/core/lib/common/nls';
+import { Deferred } from '@theia/core/lib/common/promise-util';
+import type { Mutable } from '@theia/core/lib/common/types';
+import { inject, injectable, optional } from '@theia/core/shared/inversify';
 import {
-  Port,
-  Board,
-  BoardsService,
+  OutputChannel,
+  OutputChannelManager,
+} from '@theia/output/lib/browser/output-channel';
+import {
+  BoardIdentifier,
+  boardIdentifierEquals,
+  BoardsConfig,
+  BoardsConfigChangeEvent,
   BoardsPackage,
-  AttachedBoardsChangeEvent,
-  BoardWithPackage,
+  BoardsService,
   BoardUserField,
-  AvailablePorts,
+  BoardWithPackage,
+  DetectedPorts,
+  emptyBoardsConfig,
+  isBoardIdentifier,
+  isPortIdentifier,
+  Port,
+  PortIdentifier,
+  portIdentifierEquals,
+  serializePlatformIdentifier,
 } from '../../common/protocol';
-import { BoardsConfig } from './boards-config';
-import { naturalCompare } from '../../common/utils';
-import { NotificationCenter } from '../notification-center';
-import { StorageWrapper } from '../storage-wrapper';
-import { nls } from '@theia/core/lib/common';
-import { Deferred } from '@theia/core/lib/common/promise-util';
-import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
-import { Unknown } from '../../common/nls';
+import {
+  BoardList,
+  BoardListHistory,
+  createBoardList,
+  isBoardListHistory,
+} from '../../common/protocol/board-list';
+import { Defined } from '../../common/types';
 import {
   StartupTask,
   StartupTaskProvider,
 } from '../../electron-common/startup-task';
+import { NotificationCenter } from '../notification-center';
+import { StorageWrapper } from '../storage-wrapper';
+import { BoardsDataStore } from './boards-data-store';
+
+const boardListHistoryStorageKey = 'arduino-ide:boardListHistory';
```

Comment:
probably I'm missing some context here, why do we have `camelCase` here for these variables and not "CAPS_" as we do for other module scoped `string consts`?

Reply:
> probably I'm missing some context here, why do we have `camelCase` here for these variables and not "CAPS_" as we do for other module scoped `string consts`?

Great, you brought it up. This project does not yet have a well-defined policy on naming conventions, I want to enable `@typescript-eslint/naming-convention`. If I am not mistaken, the `camelCaseWay` should be the default for constants.

This is an ongoing procedure, and I am doing it step by step. See other related [comment](https://github.com/arduino/arduino-ide/pull/2165/files#diff-44044445f3c22909b5b1649796e17e4820573f24870ebb055c2949bc88edad3aR3) in this PR.

Also, I am not introducing `namespace`s anymore, so the ESM switch should be easier.

Reply:
sounds good to me, I'm fine to adhere to `camelCase`